framework/SupervisedLearning/SyntheticHistory.py

```diff
@@ -101,7 +101,8 @@ def __trainLocal__(self, featureVals, targetVals):
     pivots = targetVals[0, :, pivotIndex]
     self.pivotParameterValues = pivots[:] # TODO any way to avoid storing these?
     residual = targetVals[:, :, :] # deep-ish copy, so we don't mod originals
-    for algo in self.tsaAlgorithms:
+    numAlgo = len(self.tsaAlgorithms)
+    for a, algo in enumerate(self.tsaAlgorithms):
```

Comment:
I don't want to be picky, but is there something more descriptive to use than `a`. Maybe `nth` or `nth_algo`? Up to you; not a showstopper.

Reply:
I'm a big fan of single letters acting as indices; I think we use this a lot in the code. It's not a code style requirement, but it is fairly common in RAVEN, and doesn't seem to cause too much confusion.

This could certainly be a bigger problem in `for` loops that are more than a handful of lines long.

Reply:
Okay haha I will from now on assume any single letter variable names are index variables in RAVEN (Tongue in cheek). But seriously, okay that's fine with me and you're right, it's a small for loop.